src-vue/lib/db/TransactionsTable.ts

```diff
@@ -0,0 +1,209 @@
+import { BaseTable, IFieldTypes } from './BaseTable';
+import { convertFromSqliteFields, toSqlParams } from '../Utils';
+import { ExtrinsicError, GenericEvent } from '@argonprotocol/mainchain';
+import { filterUndefined } from '@argonprotocol/apps-core';
+
+export enum ExtrinsicType {
+  VaultCreate = 'VaultCreate',
+  VaultModifySettings = 'VaultModifySettings',
+  VaultInitialAllocate = 'VaultInitialAllocate',
+  VaultIncreaseAllocation = 'VaultIncreaseAllocation',
+  VaultCollect = 'VaultCollect',
+  BitcoinOwnerCosignRelease = 'BitcoinOwnerCosignRelease',
+  BitcoinRequestRelease = 'BitcoinRequestRelease',
+  BitcoinInitializeLock = 'BitcoinInitializeLock',
+  Transfer = 'Transfer',
+}
+
+export enum TransactionStatus {
+  Submitted = 'Submitted',
+  InBlock = 'InBlock',
+  Finalized = 'Finalized',
+  Error = 'Error',
+  TimedOutWaitingForBlock = 'TimedOutWaitingForBlock',
+}
+
+export interface ITransactionRecord {
+  id: number; // Auto-incrementing primary key since extrinsic hash isn't implicitly unique and can overlap
+  extrinsicHash: string;
+  extrinsicJson: any;
+  extrinsicType: ExtrinsicType;
+  extrinsicMetadata: any;
+  accountAddress: string;
+  submittedAtTime: Date;
+  submittedAtBlockHeight: number;
+  submissionErrorJson: any;
+  txTip: bigint | undefined;
+  txFeePlusTip: bigint | undefined;
+  includedInBlockHeight: number | undefined;
+  includedInBlockHash: string | undefined;
+  includedInBlockTime: Date | undefined;
+  blockExtrinsicEventsJson: any[];
+  extrinsicErrorJson:
+    | { batchInterruptedIndex?: number; errorCode?: string; details?: string; message: string }
+    | undefined;
+  isFinalized: boolean;
+  status: TransactionStatus;
+  createdAt: Date;
+  updatedAt: Date;
+}
+// eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
+type ITransactionRecordKey = keyof ITransactionRecord & string;
+export class TransactionsTable extends BaseTable {
+  private bigIntFields: ITransactionRecordKey[] = ['txFeePlusTip', 'txTip'];
+  private dateFields: ITransactionRecordKey[] = ['submittedAtTime', 'includedInBlockTime', 'createdAt', 'updatedAt'];
+  private jsonFields: ITransactionRecordKey[] = [
+    'submissionErrorJson',
+    'extrinsicJson',
+    'extrinsicMetadata',
+    'extrinsicErrorJson',
+    'blockExtrinsicEventsJson',
+  ];
+  private booleanFields: ITransactionRecordKey[] = ['isFinalized'];
+
+  private get fields(): IFieldTypes {
+    return {
+      bigint: this.bigIntFields,
+      date: this.dateFields,
+      json: this.jsonFields,
+      boolean: this.booleanFields,
+    };
+  }
+
+  async fetchAll(): Promise<ITransactionRecord[]> {
+    const records = await this.db.select<any[]>('SELECT * FROM Transactions ORDER BY submittedAtBlockHeight DESC');
+    return convertFromSqliteFields(records, this.fields);
+  }
+
+  async recordInBlock(
```

Comment:
I find it helpful to explicitly mark the methods as public if they're meant to be public.